Assets/Scripts/Edges/SegmentGroup.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Assets.Scripts.Common.Extensions;
+using UnityEngine;
+
+namespace Assets.Scripts.Edges
+{
+  public class SegmentGroup
+  {
+    readonly GameObject[] _segments;
+    const float c_segmentAdditionalLength = 0.01f;
+
+    public SegmentGroup(GameObject[] segments)
+    {
+      if (segments == null || segments.Length < 1)
+        throw new ArgumentException("At least one segment has To be provided");
+
+      _segments = segments;
+    }
+
+    public GameObject MiddleSegment => _segments[_segments.Length / 2];
+
+
+    public void PlaceAlongPoints(IList<Vector3> points)
+    {
+      if (points.Count - 1 != _segments.Length)
+        throw new ArgumentException("Number of _segments and provided points don't match up");
+
+      // TODO: Consider optimizing
+      var pointsWithoutFirst = points.Skip(1);
+      var pointsWithoutLast = points.SkipLast(1);
+      var pointPairs = pointsWithoutFirst.Zip(pointsWithoutLast, (p1, p2) => (p1, p2)).ToList();
+      for (var i = 0; i < _segments.Length; i++)
+      {
+        var (p1, p2) = pointPairs[i];
+        PlaceBetweenPoints(_segments[i], p1, p2);
+      }
+    }
```

Comment:
this could be done using i index for first point and i +1 index for second second point